x-pack/plugins/security_solution/common/utils/alert_detail_path.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { addSpaceIdToPath } from '@kbn/spaces-plugin/common';
+import { ALERT_DETAILS_REDIRECT_PATH, APP_PATH } from '../constants';
+
+export const buildAlertDetailPath = ({
+  alertId,
+  index,
+  timestamp,
+}: {
+  alertId: string;
+  index: string;
+  timestamp: string;
+}) => `${ALERT_DETAILS_REDIRECT_PATH}/${alertId}?index=${index}&timestamp=${timestamp}`;
+
+export const getAlertDetailsUrl = ({
+  alertId,
+  index,
+  timestamp,
+  basePath,
+  spaceId,
+}: {
+  alertId: string;
+  index: string;
+  timestamp: string;
+  basePath?: string;
+  spaceId?: string | null;
+}) => {
+  const alertDetailPath = buildAlertDetailPath({ alertId, index, timestamp });
+  const alertDetailPathWithAppPath = `${APP_PATH}${alertDetailPath}`;
+  return basePath
```

Comment:
Consider updating the PR description to note that, as a result of this `bathPath` check, the `kibana.alert.url` feature in this PR must be enabled by adding one of the following configurations to `config/kibana.dev.yml`:

If you run kibana with `yarn start --no-base-path`:

```
server.publicBaseUrl: 'http://localhost:5601'
```

or

If you run kibana with `yarn start` (and a base path):

```
server.basePath: '/foo'
server.publicBaseUrl: 'http://localhost:5601/foo'
```

Reply:
Added already 😄